inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/InlongGroupContext.java

```diff
@@ -91,24 +93,53 @@ private void recheckState() {
         if (MapUtils.isEmpty(this.inlongStreamMap)) {
             return;
         }
+        List<StreamSource> sourcesInGroup = Lists.newArrayList();
         List<StreamSource> failedSources = Lists.newArrayList();
         this.inlongStreamMap.values().stream().forEach(inlongStream -> {
             Map<String, StreamSource> sources = inlongStream.getSources();
             if (MapUtils.isNotEmpty(sources)) {
                 for (Map.Entry<String, StreamSource> entry : sources.entrySet()) {
                     StreamSource source = entry.getValue();
-                    if (source.getState() == State.FAILED) {
-                        failedSources.add(source);
+                    if (source != null) {
+                        sourcesInGroup.add(source);
+                        if (source.getState() == State.FAILED) {
+                            failedSources.add(source);
+                        }
                     }
                 }
             }
         });
+        // check if any stream source is failed
         if (CollectionUtils.isNotEmpty(failedSources)) {
             this.state = InlongGroupState.FAILED;
             for (StreamSource failedSource : failedSources) {
                 this.groupErrLogs.computeIfAbsent("failedSources", Lists::newArrayList)
                         .add(GsonUtil.toJson(failedSource));
             }
+            return;
+        }
+        // check if any stream source is in indirect state
+        switch (this.state) {
+            case STARTED:
+                for (StreamSource source : sourcesInGroup) {
+                    if (!(source.getState() == State.NORMAL)) {
```

Comment:
Why not use `source.getState() != State.NORMAL`?

Reply:
done